File: android/core/designsystem/src/main/kotlin/com/nexussoft/verbum/designsystem/tokens/Motion.kt
package com.nexussoft.verbum.designsystem.tokens

import androidx.compose.animation.core.AnimationSpec
import androidx.compose.animation.core.FastOutSlowInEasing
import androidx.compose.animation.core.LinearOutSlowInEasing
import androidx.compose.animation.core.snap
import androidx.compose.animation.core.tween

/**
 * Animation durations and curves (docs/PRODUCT.md §41, §64).
 *
 * Motion communicates navigation through knowledge, never ambience.
 * Pass the platform's "remove animations" setting to [resolved] to honour it.
 */
object Motion {
    object Duration {
        const val QUICK_MS = 150
        const val STANDARD_MS = 250
        const val SPATIAL_MS = 450
    }

    /** State toggles, selection feedback. */
    fun <T> quick(): AnimationSpec<T> = tween(Duration.QUICK_MS, easing = LinearOutSlowInEasing)
    /** Most transitions. */
    fun <T> standard(): AnimationSpec<T> = tween(Duration.STANDARD_MS, easing = FastOutSlowInEasing)
    /** Graph expansion, focus transitions, timeline movement. */
    fun <T> spatial(): AnimationSpec<T> = tween(Duration.SPATIAL_MS, easing = FastOutSlowInEasing)

    /** Returns an instant spec when the user has animations disabled. */
    fun <T> resolved(spec: AnimationSpec<T>, reduceMotion: Boolean): AnimationSpec<T> =
        if (reduceMotion) snap() else spec
}

File: VerbumKit/Sources/DesignSystem/Tokens/Motion.swift
import SwiftUI

/// Animation durations and curves (spec §41, §64).
///
/// Motion exists to communicate navigation through knowledge — a node opening,
/// focus moving into an entity — never as ambience. Reduce Motion is honoured
/// through `resolved(_:reduceMotion:)`.
public enum Motion {
    public enum Duration {
        public static let quick: TimeInterval = 0.15
        public static let standard: TimeInterval = 0.25
        public static let spatial: TimeInterval = 0.45
    }

    /// State toggles, selection, hover-like feedback.
    public static let quick: Animation = .easeOut(duration: Duration.quick)
    /// Most transitions.
    public static let standard: Animation = .smooth(duration: Duration.standard)
    /// Graph expansion, focus transitions, timeline movement.
    public static let spatial: Animation = .snappy(duration: Duration.spatial, extraBounce: 0)

    /// Returns `nil` (no animation) when the user has Reduce Motion enabled.
    public static func resolved(_ animation: Animation, reduceMotion: Bool) -> Animation? {
        reduceMotion ? nil : animation
    }
}
